ac-55 hook up authentication to a new database schema

ExtraGravity opened the pull request:
Remove old user model.
Authentication is very similar, it just gets the user information
from the new database now. Function were written in the database layer
to get this information.

ExtraGravity committed:
ac-55 hook up authentication to a new database schema

Remove old user model.
Authentication is very similar, it just gets the user information
from the new database now. Function were written in the database layer
to get this information.

beaesguerra commented on src/database/Person/index.ts:
```diff
@@ -1,37 +1,100 @@
+function buildUserType(type: string): UserType {
+  switch (type) {
+    case UserType.ADMIN:
+      return UserType.ADMIN;
```
can you put brackets around each case?

beaesguerra commented on src/database/Person/index.ts:
```diff
@@ -1,37 +1,100 @@
+    case UserType.MONITOR:
+      return UserType.MONITOR;
+    case UserType.TRADER:
+      return UserType.TRADER;
```
you can use a default case to log the error

beaesguerra commented on src/database/Person/index.ts:
```diff
@@ -1,37 +1,100 @@
+
+/** Get attributes for a user necessary for a login from a username */
+export async function findUser(username: string): Promise<User> {
+  logger.info('Person/index.ts findUser');
```
this can be removed?

beaesguerra commented on src/database/Person/index.ts:
```diff
@@ -1,37 +1,100 @@
+  logger.info('hash, ', hashResults[0].attrvalue);
+
+  const userTypeResults = await execute<PersonCategoryName[]>(builders.getPersonCategory(uuid));
+  if (userTypeResults.length === 0) {
```
is it possible for a user to not be in a category? no right?

ExtraGravity replied:
not possible, I think it's enforced by the db.

beaesguerra commented on src/database/Person/index.ts:
```diff
@@ -1,37 +1,100 @@
+    throw new Error('Username ' + username + ' not found');
+  }
+  const uuid = uuidResults[0].personuuid;
+  logger.info('uuid, ', uuid);
```
do we need this? if we want to trackk users that are logging in, we can make this message more meaningful

beaesguerra commented on src/database/Person/index.ts:
```diff
@@ -1,37 +1,100 @@
+    case UserType.TRADER:
+      return UserType.TRADER;
+  }
+  logger.error('Unrecognized user type', type);
```
do we need to throw error?

beaesguerra commented on src/database/Person/index.ts:
```diff
@@ -1,37 +1,100 @@
+
+  const hashResults = await execute<AttrValue[]>(builders.getAttr(uuid, 'hash'));
+  if (hashResults.length === 0) {
+    throw new Error(username + ' has no password set!');
```
log this error? this seems like a pretty bad scenario we would want to know about

ExtraGravity replied:
no I think if it an error gets thrown, it either gets properly dealt with higher up, or it will logged as a crash.

beaesguerra reviewed (commented):
questions

ExtraGravity committed:
ac-55 Respond to PR comments

Throw errors instead of logging, since they'll either get handled or logged as a crash.
Adjust switch case to more best practices.

ExtraGravity committed:
Merge branch 'master' into enoch/ac-55-authentication-new-db
